virtio-devices/src/balloon.rs

```diff
@@ -78,7 +78,7 @@ pub enum Error {
 }
 
 // Got from include/uapi/linux/virtio_balloon.h
-#[repr(C, packed)]
```

Comment:
Any reason to remove packed here?

Reply:
With "packed", will got:
```
warning: borrow of packed field is unsafe and requires unsafe function or block (error E0133)
   --> virtio-devices/src/balloon.rs:336:14
    |
336 |             (&config.actual as *const _ as u64) - (&config as *const _ as u64);
    |              ^^^^^^^^^^^^^^
    |
    = note: `#[warn(safe_packed_borrows)]` on by default
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #46043 <https://github.com/rust-lang/rust/issues/46043>
    = note: fields of packed structs might be misaligned: dereferencing a misaligned pointer or even just creating a misaligned reference is undefined behavior

warning: borrow of packed field is unsafe and requires unsafe function or block (error E0133)
   --> virtio-devices/src/balloon.rs:337:56
    |
337 |         let config_actual_size = std::mem::size_of_val(&config.actual);
    |                                                        ^^^^^^^^^^^^^^
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #46043 <https://github.com/rust-lang/rust/issues/46043>
    = note: fields of packed structs might be misaligned: dereferencing a misaligned pointer or even just creating a misaligned reference is undefined behavior

   Compiling vmm v0.1.0 (/run/teawater/cloud-hypervisor/vmm)
warning: 2 warnings emitted
```
Even if I added code to support Serialize in https://github.com/teawater/cloud-hypervisor/commit/fb008ea5c46b40bb5747e88ecc1eb4971ed3ca20 , still got same warning.

And I didn't see anything use "packed".
So I remove it.

Reply:
@rbradford I was reading about `repr()` from https://doc.rust-lang.org/nomicon/other-reprs.html and I think we could remove `repr(packed)` from all our virtio config structures. Since these same structures are defined in C in the kernel, `repr(C)` should be sufficient.

Reply:
Yes, I think they should be `#[repr(C)]`

Reply:
There also another issue with packed. If any of the member field of the packed struct has aligned attribute it will give you compilation error. I guess packed is unnecessary here.